### DISCUSSION TITLE
HelloApp-macOS build issues

### DISCUSSION THREAD

**Discussion by OnSwitchOff** [unoplatform/uno#8438] [Q&A]
### Current behavior
When trying to build uno-winui-net6 "hello world" - app  the macOS fail with the error message:
```
Assembly 'Uno.UI' with identity 
'Uno.UI, Version=255.255.255.255, Culture=neutral, PublicKeyToken=null' uses 
'Microsoft.macOS, Version=12.1.303.55, Culture=neutral, PublicKeyToken=84e04ff9cfb79065'
which has a higher version than referenced assembly
'Microsoft.macOS' with identity 
'Microsoft.macOS, Version=12.1.302.122, Culture=neutral, PublicKeyToken=84e04ff9cfb79065
```

### Expected behavior
Project should build successfully

### How to reproduce it 
In a terminal
```
dotnet new unoapp-winui-net6 -o MyUnoApp
cd MyUnoApp
cd MyUnoApp.Mobile
dotnet build -t:Run -f net6.0-macOS 
```
### Prerequisites:	
- [x] macOS 12.3 (21E230)
- [x] Xcode 13.3
- [x] macos (Microsoft.NET.Sdk.macOS.Manifest-6.0.200 : 12.1.302-preview.14.122)
- [x] .NET SDK – 6.0.201
- [x] Visual Studio 2022 for Mac
- [x] Gtk3+

### NuGet package version(s) 
- Uno.WinUI Version="4.1.9"
- Uno.UI.Adapter.Microsoft.Extensions.Logging" Version="4.1.9" 
- Microsoft.Extensions.Logging Version=5.0.0
- Microsoft.Extensions.Logging.Console Version="5.0.0"
- Uno.WinUI.RemoteControl Version="4.1.9"

**Comment by jeromelaban** [comment]
Thanks for the report. Make sure to follow the directions provided here: https://github.com/unoplatform/uno/releases/tag/4.1.9, as you're likely running with an outdated version of .NET 6 Mobile preview.

**Reply by OnSwitchOff** [reply]
when I used uno.check --version 1.2.0-dev.10, it downgrade version of sdks from 14.2 to 13.5
<img width="724" alt="Screenshot 2022-03-28 at 15 57 37" src="https://user-images.githubusercontent.com/37346667/160403220-16be4b7a-5b64-4f3b-a4a6-5eb2c721bfda.png">
but I still have build error 
-You must add a reference to assembly 'Microsoft.macOS, Version=12.1.303.55....
<img width="937" alt="Screenshot 2022-03-28 at 16 05 48" src="https://user-images.githubusercontent.com/37346667/160404219-fa54b5a5-7704-4138-a9f6-69a22ed4d857.png">

**Reply by jeromelaban** [reply]
Thanks, there was a typo in the command, make sure to use `uno-check --pre` (it was an hyphen instead of `--`)

**Reply by OnSwitchOff** [reply]
In this case there isn't android-aot workload
<img width="566" alt="Screenshot 2022-03-29 at 10 22 39" src="https://user-images.githubusercontent.com/37346667/160556341-36ec40ff-41cc-4024-a3d3-09c1e3122fca.png">


Error  : The AOT compiler '' does not exist
<img width="568" alt="Screenshot 2022-03-29 at 10 31 24" src="https://user-images.githubusercontent.com/37346667/160557772-15c718d5-f4e5-49e3-8925-8b24c5094711.png">

**Reply by jeromelaban** [reply]
This is a .NET 6 Mobile issue, which is likely going to be fixed in a later preview or RC from Microsoft.

**Reply by OnSwitchOff** [reply]
Is there a timeframe for when this might happen? Will there be a new release version when this is fixed, or how can I track it down?

Thank you for information.

**Reply by jeromelaban** [reply]
We do not know, Microsoft does not provide insights there.

**Reply by dylanberry** [reply]
@OnSwitchOff this is working with all the latest/preview/RC bits (xcode, uno-check, uno templates, .NET). More info here: https://github.com/unoplatform/uno/discussions/7833

**Reply by bstrother1** [reply]
> @OnSwitchOff this is working with all the latest/preview/RC bits (xcode, uno-check, uno templates, .NET). More info here: #7833

I'm still experiencing this on Arm64 with the latest. 

🚑 Uno Platform Check v1.2.0.0 💉
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

This tool will check your Uno Platform development environment.
If problems are detected, it will offer the option to try and fix them for you, or suggest a way to fix them yourself.

─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

⏳ Synchronizing configuration... ok

⏳ Scheduling appointments... ok

🔎 OpenJDK 11.0 Checkup...

  ✔ 11.0.12 (/Library/Java/JavaVirtualMachines/microsoft-11.jdk/Contents/Home/bin/..)

🔎 Visual Studio 8.10.0 Checkup...

🔎 Android SDK Checkup...

❌ Skipped: Android Emulator

🔎 XCode 13.3 Checkup...

  ✔ Xcode.app (13.3)

🔎 .NET SDK Checkup...

  – 6.0.201 - /usr/local/share/dotnet/sdk/6.0.201

  – 6.0.202 - /usr/local/share/dotnet/sdk/6.0.202

  – 6.0.300 - /usr/local/share/dotnet/sdk/6.0.300

🔎 .NET SDK - Workload Deduplication Checkup...

🔎 .NET SDK - EnableWorkloadResolver.sentinel Checkup...

🔎 GTK3+ Checkup...

🔎 .NET SDK - Workloads (6.0.300-preview.22179.2) Checkup...

  ✔ android (Microsoft.NET.Sdk.Android.Manifest-6.0.300 : 32.0.300-rc.1.4) installed.

  ✔ ios (Microsoft.NET.Sdk.iOS.Manifest-6.0.300 : 15.4.100-rc.1.125) installed.

  ✔ maccatalyst (Microsoft.NET.Sdk.MacCatalyst.Manifest-6.0.300 : 15.4.100-rc.1.125) installed.

  ✔ tvos (Microsoft.NET.Sdk.tvOS.Manifest-6.0.300 : 15.4.100-rc.1.125) installed.

  ✔ macos (Microsoft.NET.Sdk.macOS.Manifest-6.0.300 : 12.3.100-rc.1.125) installed.

  ✔ wasm-tools (microsoft.net.workload.mono.toolchain-6.0.300 : 6.0.4) installed.

  ✔ wasm-tools (microsoft.net.workload.emscripten-6.0.300 : 6.0.4) installed.

Uno.UI is on 4.2.0-dev.673
Uno-Check is on 1.2.0-dev.34
macOS Monterey 12.3.1
